app/main.py

```diff
@@ -30,41 +30,28 @@ async def main() -> None:
         service.register_device(toilet),
     )
 
-    # create a few programs
-    wake_up_program = [
-        Message(hue_light_id, MessageType.SWITCH_ON),
-        Message(speaker_id, MessageType.SWITCH_ON),
-        Message(
-            speaker_id,
-            MessageType.PLAY_SONG,
-            "Rick Astley - Never Gonna Give You Up"
-        ),
-    ]
-
-    sleep_program = [
-        Message(hue_light_id, MessageType.SWITCH_OFF),
-        Message(speaker_id, MessageType.SWITCH_OFF),
-        Message(toilet_id, MessageType.FLUSH),
-        Message(toilet_id, MessageType.CLEAN),
-    ]
-
     # run the programs
-    await run_parallel(
-        service.send_msg(Message(hue_light_id, MessageType.SWITCH_ON)),
-        run_sequence(service.send_msg(Message(speaker_id,
-                                              MessageType.PLAY_SONG,
-                                              "Rick Astley - "
-                                              "Never Gonna Give You Up")),
-                     service.send_msg(Message(speaker_id,
-                                              MessageType.SWITCH_ON))),
-        run_parallel(service.send_msg(Message(hue_light_id,
-                                              MessageType.SWITCH_OFF)),
-                     service.send_msg(Message(speaker_id,
-                                              MessageType.SWITCH_OFF))),
-        run_sequence(service.send_msg(Message(toilet_id,
-                                              MessageType.FLUSH)),
-                     service.send_msg(Message(toilet_id,
-                                              MessageType.CLEAN))))
+    await run_sequence(
+        run_parallel(
+            service.send_msg(Message(hue_light_id, MessageType.SWITCH_ON)),
+            run_sequence(
+                service.send_msg(
+                    Message(speaker_id, MessageType.PLAY_SONG,
+                            "Rick Astley - Never Gonna Give You Up")
+                ),
+                service.send_msg(Message(speaker_id, MessageType.SWITCH_ON))
+            ),
```

Comment:
you changed the initial wake up program. you can't play song before switching on speaker